### PR TITLE
Add Support for fetching forwarded logs from third party API

### PR DESCRIPTION
**Pull request by khrm**
This enables the support for fetching logs from Third Party API. This pr enables the support for Lokistack
logs which are forwarded using vector, fluentbit or fluentd.

Other stack like ELK, splunk, etc can be added later on.

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you review them:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] [Tested your changes locally](https://github.com/tektoncd/results/blob/main/docs/DEVELOPMENT.md) (if this is a code change)
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user-facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contain the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Enables the support for fetching logs from Third Party API. We have enabled the support for Loki for now. Later on, support for other logging stacks like Splunk, Google Cloud Logging, etc can be added.
```

### CODE REVIEW

**Review by khrm — COMMENTED** [review]
/kind feature

File: cmd/watcher/main.go
Comment (khrm) [inline cmd/watcher/main.go:65]: This is just temporarily for testing purpose.

**Comment by khrm** [comment]
/hold
Wait for Pipelines release.

File: pkg/apis/v1alpha3/types.go
Comment (enarha) [inline pkg/apis/v1alpha3/types.go:57]: Just a note. I think it makes sense to make all these values case insensitive. It's more user friendly and there is no chance of name collisions.

File: docs/logging-support.md
Comment (enarha) [inline docs/logging-support.md:6]:
```suggestion
Tekton Results supports fetching logs from third party logging APIs.
```

File: pkg/apis/v1alpha3/types.go
Comment (khrm) [reply pkg/apis/v1alpha3/types.go:57]: Yes. I was thinking of using strings.lower so that a user can send anything.

File: cmd/api/main.go
Comment (khrm) [reply cmd/api/main.go:308]: Yes. It should be different. I didn't test HTTP API. Only GRPC. That's why I didn't find it.

File: pkg/api/server/v1alpha2/server.go
Comment (khrm) [reply pkg/api/server/v1alpha2/server.go:166]: I was taking from the config earlier but I found that it's not needed.
Let's say someone chooses LOGS_API = true and LOGS_TYPE = loki. It means he is choosing the plugin API not the older API.

Comment (khrm) [reply pkg/api/server/v1alpha2/server.go:166]: I will add a comment above to explain this.

File: pkg/api/server/v1alpha2/server.go
Comment (khrm) [reply pkg/api/server/v1alpha2/server.go:180]: This value is taken from Retention Policy Config. Maybe I should document this also.
If the value is incorrect, I think we shouldn't stop the server. That's why I set the default. Maybe I should put `errorf` instead of `warnf`.

Comment (enarha) [reply pkg/api/server/v1alpha2/server.go]: IMO we should fail. The worst thing is to lie to the admin user who set that value. If we do not fail and instead of silently change the value, the admin will think everything was ok and the system is running with the intended configuration while in fact it is not. @vdemeester WDYT?

File: pkg/api/server/v1alpha2/server.go
Comment (enarha) [reply pkg/api/server/v1alpha2/server.go:170]: If I set LOGS_API = true and LOGS_TYPE = loki and then decide to disable the logs functionality by changing LOGS_API = false and keep LOGS_TYPE = loki what will happen? We set the LOGS_API to true from code?

File: cmd/api/main.go
Comment (khrm) [reply cmd/api/main.go]: Done.

File: pkg/api/server/v1alpha2/server.go
Comment (khrm) [reply pkg/api/server/v1alpha2/server.go]: We are failing the Retention Policy Agent, so the admin will be aware of this. But now that I think, it's possible someone doesn't monitor Agent but only API and Watcher. So we should fail that also.

**Comment by khrm** [comment]
image
Done with testing GRPC endpoints. Now testing Rest API end point.

**Comment by khrm** [comment]
REST API endpoints are also working fine: 
image

File: docs/logging-support.md
Comment (avinal) [inline docs/logging-support.md:1]: Why the content of this file is in code block? Is this intentional?

**Review by avinal — APPROVED** [review]
Just one comment. Rest looks okay to me.

File: docs/logging-support.md
Comment (khrm) [reply docs/logging-support.md:1]: Good catch. Let me change that.

**Comment by enarha** [comment]
@khrm I have no other comments and I'm ready to lgtm this change. You mentioned above you are going to add another update based on Avinal's comment. Once you are ready, I'll lgtm. Otherwise it'll be removed on update or merge automatically before.

**Comment by khrm** [comment]
@enarha I have updated with that change.

**Comment by khrm** [comment]
https://github.com/tektoncd/results/compare/802dda3b5d5e1b5f0b8ae2f9495ba692d714d69a..7b361fb4536eab682abd461a9adf42280029065b

File: docs/logging-support.md
Comment (khrm) [reply docs/logging-support.md:1]: Somehow it looks good in rendering. The editor was configured to create a new md  file as runbook.

Comment (khrm) [reply docs/logging-support.md:1]: I have changed it.

**Pull request merged**
